src/main/java/dev/galacticraft/api/universe/celestialbody/landable/Landable.java

```diff
@@ -26,43 +26,34 @@
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.world.level.Level;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Optional;
 
 /**
- * Represents a {@link dev.galacticraft.api.universe.celestialbody.CelestialBodyType<C> celestial body type} that has a {@link Level} linked to itself.
+ * Represents a {@link dev.galacticraft.api.universe.celestialbody.CelestialBodyType<C> celestial body type} that can potentially have a {@link Level} linked to itself.
  *
  * @param <C> the type of configuration
  */
 public interface Landable<C extends CelestialBodyConfig> {
+    default boolean hasLinkedWorld(C config) {
+        return world(config) != null;
+    }
+
     /**
-     * Returns the registry key of the {@link Level} this celestial body is linked to
+     * Returns the registry key of the {@link Level} this celestial body is linked to, or {@code null} if it does not exist (say, if it's a decorated planet)
      *
      * @param config the celestial body configuration to be queried
      * @return the registry key of the {@link Level} this celestial body is linked to
      */
-    @NotNull ResourceKey<Level> world(C config);
+    @Nullable ResourceKey<Level> world(C config);
```

Comment:
If a celestial body is `Landable` you should be able to land on it. If you don't want a landable celestial body, don't use a landable type.